src/components/member-role-update/index.js

```diff
@@ -26,48 +23,110 @@ const MemberRoleUpdate = () => {
     `${BASE_API_URL}/users/${selectedMember}`
   );
 
+  const member = userData?.user.roles.member;
+  const archived = userData?.user.roles.archived;
+  const userId = userData?.user.id;
   const { data: tagData } = useFetch(`${BASE_API_URL}/tags`);
   const { data: levelData } = useFetch(`${BASE_API_URL}/levels`);
 
-  const promoteToMember = async (user) => {
+  const promoteToMember = async (id) => {
     setIsUpdating(true);
-    const { status } = await moveToMember(user);
+    const role = {
+      member: true,
+    };
+    const { status } = await memberRoleUpdate(id, role);
     setIsUpdating(false);
-    if (status === 204) {
-      setUpdateStatus('User moved to member');
+    if (status === 200) {
+      setUpdateStatus('user promoted to a member');
+    } else {
+      setUpdateStatus('some error occured, please contact admin');
+    }
+  };
+
+  const demoteFromMember = async (id) => {
+    setIsUpdating(true);
+    const role = {
+      member: false,
+    };
+    const { status } = await memberRoleUpdate(id, role);
+    setIsUpdating(false);
+    if (status === 200) {
+      setUpdateStatus('user deomoted from member');
+    } else {
+      setUpdateStatus('some error occured, please contact admin');
+    }
+  };
+
+  const archiveTheMember = async (id) => {
+    setIsUpdating(true);
+    const role = {
+      archived: true,
+    };
+    const { status } = await memberRoleUpdate(id, role);
+    setIsUpdating(false);
+    if (status === 200) {
+      setUpdateStatus('user archived!');
     } else {
       setUpdateStatus('Some error occured, please contact admin');
     }
   };
-  const archiveTheMember = async (user) => {
+
+  const unArchiveMember = async (id) => {
     setIsUpdating(true);
-    const { status } = await archiveMember(user);
+    const role = {
+      archived: false,
+    };
+    const { status } = await memberRoleUpdate(id, role);
     setIsUpdating(false);
-    if (status === 204) {
-      setUpdateStatus('User archived!');
+    if (status === 200) {
+      setUpdateStatus('user un-archived!');
     } else {
       setUpdateStatus('Some error occured, please contact admin');
     }
   };
 
+  const memberRoleUpdateButton = member ? (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => demoteFromMember(userId)}
+    >
+      Demote Member
+    </button>
+  ) : (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => promoteToMember(userId)}
+    >
+      Promote to Member
+    </button>
+  );
+
+  const memeberArchiveUnArchiveButton = archived ? (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => unArchiveMember(userId)}
+    >
+      Unarchive Member
+    </button>
+  ) : (
+    <button
+      className={classNames.moveToMember}
+      type="button"
+      onClick={() => archiveTheMember(userId)}
+    >
+      Archive Member
+    </button>
+  );
```

Comment:
as mentioned above, can we not just check for the status of the user, and then on the basis of that change the payload and CTA text?